File: src/005/problem005.rs
fn greatest_common_divisor(a: i32, b: i32) -> i32 {
    let mut a = a;
    let mut b = b;
    while b != 0 {
        let module = a % b;
        a = b;
        b = module;
    }
    a
}

fn least_common_multiple(a: i32, b: i32) -> i32 {
    a * b / greatest_common_divisor(a, b)
}

fn solve() -> i32 {
    (1..20).fold(1, |acc, x| least_common_multiple(acc, x))
}

fn main() {
    let result = solve();
    println!("{}", result);
}

File: src/005/problem005.swift
func greatestCommonDivisor(_ x: Int, _ y: Int) -> Int {
    var x = x
    var y = y
    while y != 0 {
        let mod = x % y
        x = y
        y = mod
    }
    return x
}

func leastCommonMultiple(_ x: Int, _ y: Int) -> Int {
    return x * y / greatestCommonDivisor(x, y)
}

func solve() -> Int {
    return (1...20).reduce(1) { sum, x in leastCommonMultiple(sum, x) }
}

let result = solve()
print(result)
